harvester/src/main/java/com/github/reflectoring/infiniboard/harvester/scheduling/SchedulingService.java

```diff
@@ -31,6 +43,13 @@ public SchedulingService() throws SchedulerException {
         this.scheduler = scheduler;
     }
 
+    synchronized public void registerJob(String type, Class<? extends SourceJob> clazz) {
+        if (jobMap.containsKey(type)) {
+            throw new RuntimeException(String.format("job type %s is already registered by %s", type, jobMap.get(type)));
```

Comment:
We should discuss exception handling at a meeting. The intention of the code is to fail at application level if jobs cannot be registered.